scripts/release.main.kts

```diff
@@ -51,14 +51,31 @@ fun getCurrentVersion(): String {
     return matchResult.groupValues[1]
 }
 
-fun getNext(version: String, position: Int) = version.split(".").mapIndexed { index, s ->
-    when {
-        index == position -> (s.toInt() + 1).toString()
-        index > position -> "0"
-        else -> s
+val versionRegex = Regex("(?<major>[0-9]+)\\.(?<minor>[0-9]+)\\.(?<patch>[0-9]+)(-(?<variantName>alpha|beta)\\.(?<variant>[0-9]+))?(-SNAPSHOT)?")
+fun getNext(version: String, position: Int): String {
+    val groupName = when (position) {
+        0 -> "major"
+        1 -> "minor"
+        2 -> "patch"
+        3 -> "variant"
+        else -> throw IllegalArgumentException("position must be 0, 1, 2 or 3")
     }
-}.joinToString(".")
+    return version.replace(versionRegex) {
+        when (groupName) {
+            "major" -> "${it.groups["major"]!!.value.toInt() + 1}.0.0"
+            "minor" -> "${it.groups["major"]!!.value}.${it.groups["minor"]!!.value.toInt() + 1}.0"
+            "patch" -> "${it.groups["major"]!!.value}.${it.groups["minor"]!!.value}.${it.groups["patch"]!!.value.toInt() + 1}"
+            "variant" -> {
+                val variantName = it.groups["variantName"]?.value ?: "alpha"
+                val newVariant = (it.groups["variant"]?.value?.toInt() ?: 0) + 1
+                "${it.groups["major"]!!.value}.${it.groups["minor"]!!.value}.${it.groups["patch"]!!.value}-$variantName.$newVariant"
+            }
```

Comment:
NitPick: it's called a "pre-release" in [SemVer](https://semver.org/) terms. Variant got me confused for a minute because we have Android variants and Gradle variants already.